gateway/middleware.go

```diff
@@ -0,0 +1,53 @@
+package gateway
+
+import (
+	"log"
+	"net/http"
```

Comment:
This is fine for now, cortex and thanos use https://github.com/weaveworks/common and https://github.com/gorilla/mux.  This last one is archived.

Let's see how far we get with the standard library.